File: 1928.py
def test(num):
    print(num)
    if num == 1:
        print(end='')
    elif (num % 2) == 1:
        test(3 * num + 1)
    else:
        test(int(num / 2))


num = int(input())

test(num)

File: 1928.cpp
#include <iostream>
using namespace std;

void test(int num)
{
    cout << num << endl;
    if (num == 1)
    {
    }
    else if (num % 2 == 1)
    {
        test(3 * num + 1);
    }
    else
    {
        test(num / 2);
    }
}
int main()
{
    int num;
    cin >> num;

    test(num);

    return 0;
}